smartsim/_core/utils/telemetry/telemetry.py

```diff
@@ -524,6 +317,272 @@ async def on_timestep(self, timestamp: int) -> None:
                     await self._to_completed(timestamp, completed_entity, step_info)
 
     async def shutdown(self) -> None:
+        """Release all resources owned by the `ManifestEventHandler`"""
         logger.debug(f"{type(self).__name__} shutting down...")
-        await self._collector.shutdown()
+        await self._collector_mgr.shutdown()
         logger.debug(f"{type(self).__name__} shutdown complete...")
+
+
+class TelemetryMonitorArgs:
+    """Strongly typed entity to house logic for validating
+    configuration passed to the telemetry monitor"""
+
+    def __init__(
+        self,
+        exp_dir: str,
+        frequency: int,
+        cooldown: int,
+        exp_id: str = "",
+        log_level: int = logging.DEBUG,
+    ) -> None:
+        """Initialize the instance with inputs and defaults
+
+        :param exp_dir: root path to experiment outputs
+        :type exp_dir:  str
+
+        :param frequency: desired frequency of metric & status updates (in seconds)
+        :type frequency:  int
+
+        :param frequency: cooldown period (in seconds) before automatic shutdown
+        :type frequency:  int
+
+        :param exp_id: unique experiment ID to support multi-experiment drivers
+        :type exp_id:  str, see `create_short_id_str`
+
+        :param log_level: log level to apply to python logging
+        :type log_level: logging._Level"""
+        self.exp_dir: str = exp_dir
+        self.frequency: int = frequency  # freq in seconds
+        self.cooldown: int = cooldown  # cooldown in seconds
+        self.exp_id: str = exp_id
+        self.log_level: int = log_level
+        self._max_frequency = 600
+        self._validate()
+
+    @property
+    def min_frequency(self) -> int:
+        """The minimum duration (in seconds) for the monitoring loop to wait
+        between executions of the monitoring loop. Shorter frequencies may
+        not allow the monitoring loop to complete. Adjusting the minimum frequency
+        can result in inconsistent or missing outputs due due to the telemetry
+        monitor cancelling processes that exceed the allotted frequency."""
+        return 1
+
+    @property
+    def max_frequency(self) -> int:
```

Comment:
max frequency is the top end of the range for the validation in `def _check_frequency(self) -> None:`. This should be ok, since the frequency is really given by the user (or more likely, just the smartsim driver that starts it)